contrib/fuzz/unstable_call_graph.py

```diff
@@ -0,0 +1,165 @@
+    )
+    parser.add_argument(
+        "--binary-file",
+        help="Path to binary file from which to generate the call graph",
+        type=str,
+        required=True)
+    parser.add_argument(
+        "--unstable-functions-file",
+        help="Path to file that contains the unstable functions",
+        type=str,
+        required=True)
+    parser.add_argument(
+        "--target-function",
+        help="The target function that is the top of the call graph (e.g. fuzz_target_name_fuzz_target)",
+        type=str,
+        required=True)
+    parser.add_argument(
+        "--skip-strings-file",
+        help="Path to file that contains strings that if at the start of a function name, that function is skipped",
+        type=str,
+        default=None
+    )
+
+    args = parser.parse_args()
+    binary_file = args.binary_file
+    unstable_functions_file = args.unstable_functions_file
+    skip_strings_file = args.skip_strings_file
+    target_function = args.target_function
+    
+    with open(unstable_functions_file, 'r') as unstable_file:
+        unstable_functions = set(line.strip() for line in unstable_file if line.strip())
+
+    strings_to_skip = set()
+    if skip_strings_file:
+        with open(skip_strings_file, 'r') as skip_file:
+            strings_to_skip = set(line.strip() for line in skip_file if line.strip())
+
+    try:
+        lines = check_output(["objdump", "-d", args.binary_file]).splitlines()
+    except:
+        exit()
+
+    G = nx.DiGraph()
+    curFunc = None
+
+    for l in lines:
+        l = l.decode('utf-8')
+        m = re.match(r'^([0-9a-zA-Z]+)\s+<(.*)>:$', l)
+        if m:
+            curFunc = m.group(2)
+            continue
+
+        if curFunc == None:
+            continue
+
+        m = re.match(r'^.*\bcall\s+([0-9a-zA-Z])+\s+<(.*)>$', l)
+        if m:
+            G.add_edge(curFunc, m.group(2))
+    
+    seen_edges = set()
+
+    def create_hierarchy(G, root):
+        demangled = demangle(root)
+        node = {
+            demangled: {
+                "children": {},
+                "flag": root in unstable_functions,
+            }
+        }
+        for child in G.successors(root):
+            edge = (root, child)
+            if edge in seen_edges:
+                continue
+            seen_edges.add(edge)
+            
+            demangled_child = demangle(child)
+            if child in unstable_functions and any(demangled_child.startswith(string) for string in strings_to_skip):
+                continue            
+            
+            node[demangled]["children"] = node[demangled]["children"] | create_hierarchy(G, child)
+        return node
+    
+    
+    target_nodes = [node for node in G.nodes() if target_function in node]
+    hierarchy = {}
+    
+    # Do we really need the for loop here or should we just do hierarchy = hierarchy | create_hierarchy(G, target_node)
+    for root in target_nodes:
+        hierarchy = hierarchy | create_hierarchy(G, root)
```

Comment:
I think we should do what your comment suggests here, i.e. look for the "target_function" node (if it doesn't exist, exit with an error) and only build the hierarchy for that node.

Reply:
Yes that looks fine but I think the script should exit with an error if the target function is not found.